Src/Metrics/MetricTags.cs

```diff
@@ -9,50 +8,41 @@ namespace Metrics
     /// </summary>
     public struct MetricTags : Utils.IHideObjectMembers
     {
-        private static readonly string[] empty = new string[0];
+        private static readonly KeyValuePair<string,string>[] empty = Enumerable.Empty<KeyValuePair<string, string>>().ToArray();
 
-        public static readonly MetricTags None = new MetricTags(Enumerable.Empty<string>());
+        public static readonly MetricTags None = new MetricTags(Enumerable.Empty<KeyValuePair<string, string>>());
 
-        private readonly string[] tags;
+        private readonly KeyValuePair<string, string>[] tags;
 
-        public MetricTags(params string[] tags)
+        public MetricTags(params KeyValuePair<string, string>[] tags)
         {
             this.tags = tags.ToArray();
         }
 
-        public MetricTags(IEnumerable<string> tags)
-            : this(tags.ToArray())
+        public MetricTags(IEnumerable<KeyValuePair<string, string>> tags) : this(ToTags(tags).ToArray())
         { }
 
-        public MetricTags(string commaSeparatedTags)
-            : this(ToTags(commaSeparatedTags))
-        { }
-
-        public string[] Tags
+        public KeyValuePair<string, string>[] Tags
         {
             get
             {
-                return tags ?? empty;
+                return this.tags ?? empty;
             }
         }
 
-        private static IEnumerable<string> ToTags(string commaSeparatedTags)
+        private static IEnumerable<KeyValuePair<string, string>> ToTags(IEnumerable<KeyValuePair<string, string>> tags)
```

Comment:
Consider rename to 'ValidateTags' / 'CleanTags' or 'FormatTags' since this method is technically not converting ToTags anymore but rather putting the tags in an expected format.